sshd-sftp/src/main/java/org/apache/sshd/client/subsystem/sftp/SftpRemotePathChannel.java

```diff
@@ -262,7 +262,7 @@ public long transferTo(long position, long count, WritableByteChannel target) th
                 while (totalRead < count) {
                     int read = sftp.read(handle, curPos, buffer, 0, buffer.length);
                     if (read > 0) {
-                        ByteBuffer wrap = ByteBuffer.wrap(buffer);
+                        ByteBuffer wrap = ByteBuffer.wrap(buffer, 0,  (int) Math.min(count - totalRead, buffer.length));
```

Comment:
Shouldn't this be ?

```java
ByteBuffer wrap = ByteBuffer.wrap(buffer, 0,  read);
```

Reply:
Both are same, but to make it understand easily it is better to change it to read.
 I have tested with read and it working fine. Committed the change to forked branch